keps/sig-architecture/960-conformance-behaviors/README.md

```diff
@@ -118,35 +142,236 @@ tests and test scaffolding to quickly cover those behaviors.
+<<[/UNRESOLVED]>>
+
+<<[UNRESOLVED @spiffxp: should this capture preconditions for testing: ]>>
+* All behaviors present
+* All behaviors covered by tests
+* Tests should have been around to verify non-flakiness
+* Tests should have been reviewed by conformance reviewer to make sure they meet
+  the criteria - can we front load this?
+<<[/UNRESOLVED]>>
+
+##### Creating a brand new feature, either required or optional
+During creation of an alpha or beta feature, conformance tests are not required,
+nor are conformance behaviors. However, at the beta stage, the expectation
+should be to have some quality end-to-end tests, and so we may want to allow the
+definition of the behaviors at that time too. Tasks then would be similar to
+some of those for GA promotion:
+1. Define expected behaviors
+1. Get provisional behaviors approved by the conformance-behavior-approvers
+1. Write tests to cover those behaviors
+
+<<[UNRESOLVED]>>
+@johnbelamaric
+Ideally we could avoid the provisional behavior approval. Maybe we can have a
+way to have a separate behaviors area for beta stuff? Or maybe we just don't
+have this at all for beta, and it waits till GA. The reason I bring up making
+bahaviors now is because the initial idea of `kubetestgen` was to support these
+steps: creation of behaviors, and creation of standard e2e tests for those
+behaviors.
+@jefftree
+Getting this list of behaviors approved is something that needs to eventually be
+done before hitting GA. I don't know how much these behaviors would change
+between beta and GA, but if they're relatively stable and mainly additive,
+starting the process early seems fine. Similar to your previous point, we should
+look to move some of these behavior approvals earlier in the process to avoid
+the chaos of reviews when a feature is going to GA.
+<<[/UNRESOLVED]>>
+
+#### Role: Kubernetes Vendor
+
+##### Evaluating a distribution for conformance
+* Must set up test cluster and run sonobuoy conformance tests
+* If successful, submit PR to CNCF. If failures exist, debug them
+
+##### Identifying the profiles supported by a distribution
+* Must run a set of conformance tests for each profile supported
+
+#### Role: CNCF Conformance Program
+
+##### Evaluate a Vendor Submission
+* Must confirm that the version of the tests being run matches the version being
+  certified
+* Must confirm the set of tests being run matches the set of tests for the
+  version (+ profile(s)) being certified
+* Must confirm that all behaviors are covered by a test that executes, and that
+  no tests fail (This isn’t done today: verify skew policy - confirm a cluster
+  being certified for version 1.x also passes conformance tests for version
+  1.x-1)
+
+#### Role: CI Job
+
+##### Identify a PR as requiring conformance review
+PR must touch file in conformance-specific directory
+* eg: update test/conformance/behaviors/..
+* eg: mv from test/e2e to test/conformance
+
+##### Evaluating a PR for conformance coverage
```

Comment:
Currently planning to continue the PR for conformance coverage with endpoints until we have this behaviours list.